### PR TITLE
Change type parameters to MapType in lang.table

### LINKED ISSUE #30608: Fix incorrect `table` type usages in `lang.table`
**Description:**
Refer https://github.com/ballerina-platform/ballerina-spec/issues/849

### PR DESCRIPTION
**Pull request by gimantha**
Fixes #30608

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java [hunk 1]
```diff
@@ -273,6 +273,10 @@ private BType createBuiltInType(BType type, Name name, long flags) {
                 anydataType.name = name;
                 anydataType.flags |= flags;
                 return anydataType;
+            case TypeTags.MAP:
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java:276]: Not sure if we can consider map as a built-in type in this context based on the other types we are handling here. Do we know where it breaks if we do not add this here?

Comment (gimantha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java]: yes. we can remove it. i ll remove it

File: langlib/lang.table/src/main/ballerina/table.bal [hunk 2]
```diff
@@ -160,7 +166,7 @@ public isolated function reduce(table<Type> t, @isolatedParam function(Type1 acc
 # + t - the table
 # + k - the key
 # + return - the member of parameter `t` that had key parameter `k`
-public isolated function remove(table<Type> key<KeyType> t, KeyType k) returns Type = @java:Method {
+public isolated function remove(table<MapType> key<KeyType> t, KeyType k) returns MapType = @java:Method {
```
Comment (lasinicl) [inline langlib/lang.table/src/main/ballerina/table.bal:169]:
```ballerina
type MapType map<any|error>;
type KeyType anydata;

public function main() {
    table<MapType> key<KeyType> t = table [];
}
```
Specifying a `key-type-constraint` for a `table` constrained with a `map` type results in `table with constraint of type map cannot have key specifier or key type constraint` compilation error. Shouldn't the same be applicable here when such `table` type is given as a function param?

Comment (lasinicl) [reply langlib/lang.table/src/main/ballerina/table.bal:169]:
```ballerina
function foo(table<MapType> key<KeyType> t) {}
```
This should not be allowed right? This doesn't result in a compilation error ATM.

Comment (gimantha) [reply langlib/lang.table/src/main/ballerina/table.bal:169]: hmm. yeah. I think we will have to handle it specifically, by checking if it is not a langlib function. will send the fix with this pr

**Review by lasinicl — APPROVED** [review]

**Pull request merged**